src/components/forms/DMGPalettePicker.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useCallback } from "react";
+import {useAppSelector } from "store/hooks";
+import { NumberInput } from "ui/form/NumberInput";
+import { castEventToInt } from "renderer/lib/helpers/castEventValue";
+import styled from "styled-components";
+import type { MonoPalette } from "shared/lib/entities/entitiesTypes";
+
+type DMGPalettePickerProps = {
+  name: string
+  palette: MonoPalette
+  isSpritePalette: boolean
+  onChange: (palette: MonoPalette) => void;
+  showName?: boolean
+};
+
+export const DMGPalettePicker = ({
+  name,
+  palette,
+  isSpritePalette,
+  onChange,
+  showName
+}: DMGPalettePickerProps) => {
+
+  const settings = useAppSelector((state) => state.project.present.settings);
+  const dmgColors = [settings.customColorsWhite, settings.customColorsLight, settings.customColorsDark, settings.customColorsBlack];
+  const fields = isSpritePalette ? [1, 2, 3] : [0, 1, 2, 3];
```

Comment:
At one point, I was showing all 4 input boxes for sprites. Then, I realized the lowest two bits are never used anyway and it felt silly to show them. I didn't want to change the underlying data structure from `[number, number, number, number]` to `number[]` though because dealing with arrays of different sizes complicated the logic further and didn't line up as well with the byte that it represented in a build.